core-rust/state-manager/src/state_computer.rs

```diff
@@ -1270,6 +1268,58 @@ where
         })
     }
 
+    fn verify_post_commit_protocol_version(
+        &self,
+        commit_state_version: StateVersion,
+        new_protocol_state: &ProtocolState,
+        next_version_from_local_execution: &Option<ProtocolVersionName>,
+        next_version_from_proof: &Option<ProtocolVersionName>,
+    ) {
+        match (next_version_from_local_execution, next_version_from_proof) {
+            (Some(local_version), Some(proof_version)) if local_version == proof_version => {
+                // All good, local execution enacts the same protocol version as proof. No-op.
+            }
+            (Some(local_version), Some(proof_version)) => {
+                // We're enacting a different version than the proof
+                panic!(
+                    "FATAL: At state version {} this node enacts {}, but this doesn't match the proof, which enacts {} (i.e. the validator set have enacted a different version). Make sure your node is running the latest software version.",
```

Comment:
To avoid confusion, can we change this to "this node wants to enact" and make clear we're terminating without enacting anything?

I want to avoid node runners being concerned that they think their ledger has corrupted from this.